phoenix-core/src/it/java/org/apache/phoenix/end2end/ArrayRemoveFunctionIT.java

```diff
@@ -0,0 +1,383 @@
+		assertTrue(rs.next());
+
+		assertEquals(conn.createArrayOf("INTEGER", new Integer[] { 2345, 46345, 23234, 456 }), rs.getArray(1));
+		assertFalse(rs.next());
+	}
+
+	@Test
+	public void testArrayRemoveFunctionFirstElement() throws Exception {
+		ResultSet rs = conn.createStatement().executeQuery(
+				"SELECT ARRAY_REMOVE(varchars,'2345') FROM " + tableName + " WHERE region_name = 'SF Bay Area'");
+		assertTrue(rs.next());
+
+		assertEquals(conn.createArrayOf("VARCHAR", new String[] { "46345", "23234" }), rs.getArray(1));
+		assertFalse(rs.next());
+	}
+
+	@Test
+	public void testArrayRemoveFunctionMiddleElement() throws Exception {
+		ResultSet rs = conn.createStatement().executeQuery(
+				"SELECT ARRAY_REMOVE(varchars,'46345') FROM " + tableName + " WHERE region_name = 'SF Bay Area'");
+		assertTrue(rs.next());
+
+		assertEquals(conn.createArrayOf("VARCHAR", new String[] { "2345", "23234" }), rs.getArray(1));
+		assertFalse(rs.next());
+	}
+
+	@Test
+	public void testArrayRemoveFunctionLastElement() throws Exception {
+		ResultSet rs = conn.createStatement().executeQuery(
+				"SELECT ARRAY_REMOVE(varchars,'23234') FROM " + tableName + " WHERE region_name = 'SF Bay Area'");
+		assertTrue(rs.next());
+
+		assertEquals(conn.createArrayOf("VARCHAR", new String[] { "2345", "46345" }), rs.getArray(1));
+		assertFalse(rs.next());
+	}
+
+	@Test
+	public void testArrayRemoveFunctionOneElement() throws Exception {
+		ResultSet rs = conn.createStatement().executeQuery(
+				"SELECT ARRAY_REMOVE(oneItem,'alone') FROM " + tableName + " WHERE region_name = 'SF Bay Area'");
+		assertTrue(rs.next());
+
+		assertEquals(conn.createArrayOf("VARCHAR", new String[0]), rs.getArray(1));
+		assertFalse(rs.next());
+	}
+
+	@Test
+	public void testArrayRemoveFunctionRepeatingElements() throws Exception {
+		ResultSet rs = conn.createStatement().executeQuery(
+				"SELECT ARRAY_REMOVE(chars,'c') FROM " + tableName + " WHERE region_name = 'SF Bay Area'");
+		assertTrue(rs.next());
+
+		assertEquals(conn.createArrayOf("CHAR", new String[] { "a", "bbbb", "ddd", "e" }), rs.getArray(1));
+		assertFalse(rs.next());
+	}
+
+	@Test
+	public void testArrayRemoveFunctionCharFromVarcharArray() throws Exception {
+		ResultSet rs = conn.createStatement().executeQuery(
+				"SELECT ARRAY_REMOVE(varchars,char2) FROM " + tableName + " WHERE region_name = 'SF Bay Area'");
+		assertTrue(rs.next());
+
+		assertEquals(conn.createArrayOf("VARCHAR", new String[] { "46345", "23234" }), rs.getArray(1));
+		assertFalse(rs.next());
+	}
+
+	@Test
+	public void testArrayRemoveFunctionVarcharFromCharArray() throws Exception {
+		ResultSet rs = conn.createStatement().executeQuery(
+				"SELECT ARRAY_REMOVE(chars,varchar1) FROM " + tableName + " WHERE region_name = 'SF Bay Area'");
+		assertTrue(rs.next());
+
+		assertEquals(conn.createArrayOf("CHAR", new String[] { "a", "c", "ddd", "e", "c" }), rs.getArray(1));
```

Comment:
Instead of removing "bbbb" which is the longest element so won't have trailing spaces, try removing "a" or "c" as it's a better test.